devicetypes/erocm123/inovelli-2-channel-smart-plug-mcd.src/inovelli-2-channel-smart-plug-mcd.groovy

```diff
@@ -0,0 +1,332 @@
+	])
+}
+
+void childOn(String dni) {
+	log.debug "childOn($dni)"
+	def cmds = []
+
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.basicV1.basicSet(value: 0xFF), channelNumber(dni))))
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+
+	sendHubCommand(cmds, 1000)
+}
+
+void childOff(String dni) {
+	log.debug "childOff($dni)"
+	def cmds = []
+
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.basicV1.basicSet(value: 0x00), channelNumber(dni))))
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+
+	sendHubCommand(cmds, 1000)
+}
+
+void childRefresh(String dni) {
+	log.debug "childRefresh($dni)"
+	def cmds = []
+
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+
+	sendHubCommand(cmds, 1000)
+}
+
+def poll() {
+	log.debug "poll()"
+
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh()"
+
+	commands([
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 1),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), 2),
+	])
+}
+
+def ping() {
+	log.debug "ping()"
+
+	refresh()
+}
+
+def getCheckInterval() {
+	2 * 15 * 60 + 2 * 60
+}
+
+def installed() {
+	log.debug "installed()"
+
+	sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	createChildDevices()
+	response(refresh())
+}
+
+def updated() {
+	log.debug "updated()"
+
+	if (!childDevices) {
+		createChildDevices()
+	} else if (device.label != state.oldLabel) {
+		childDevices.each {
+			if (it.label == "${state.oldLabel} (CH${channelNumber(it.deviceNetworkId)})") {
+				def newLabel = "${device.displayName} (CH${channelNumber(it.deviceNetworkId)})"
+				it.setLabel(newLabel)
+			}
+		}
+		state.oldLabel = device.label
+	}
+
+	sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	log.debug "${device.displayName} parameter '${cmd.parameterNumber}' with a byte size of '${cmd.size}' is set to '${cmd2Integer(cmd.configurationValue)}'"
+}
+
+private encap(cmd, endpoint) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+private command(physicalgraph.zwave.Command cmd) {
+	if (state.sec) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private commands(commands, delay = 1000) {
+	delayBetween(commands.collect {
+		command(it)
+	}, delay)
+}
+
+private channelNumber(String dni) {
+	dni.split(":")[-1] as Integer
+}
+
+private void createChildDevices() {
+	state.oldLabel = device.label
```

Comment:
I think this needs to be saving `device.displayName` instead of `device.label`. The `label` isn't set when the device is created so `displayName` returns `device.name` instead. As written right now updating the label on the device doesn't update the label on the children.

Reply:
Turns out that it was the if statement in the closure for updating each child device. I removed it and the code appears to work now.